{
  Statement stmt=con.createStatement();
  boolean exceptionFound=false;
  try {
    stmt.execute(sql);
  }
 catch (  SQLException e) {
    assertTrue("Adequate error messaging not found for '" + sql + "': "+ e.getMessage(),e.getMessage().contains(expectedMessage));
    assertEquals("Expected SQLState not found for '" + sql + "'",expectedSQLState,e.getSQLState());
    assertEquals("Expected error code not found for '" + sql + "'",expectedErrorCode,e.getErrorCode());
    exceptionFound=true;
  }
  assertNotNull("Exception should have been thrown for query: " + sql,exceptionFound);
  stmt.close();
}
